src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkRuleConfiguredTargetUtil.java

```diff
@@ -643,8 +658,13 @@ private static void addSimpleProviders(
       RunfilesSupport runfilesSupport = null;
       if (!computedDefaultRunfiles.isEmpty()) {
         Preconditions.checkNotNull(executable, "executable must not be null");
-        runfilesSupport =
-            RunfilesSupport.withExecutable(ruleContext, computedDefaultRunfiles, executable);
+        if (starlarkArgsProvided) {
+          runfilesSupport = RunfilesSupport.withExecutableNoArgs(
+              ruleContext, computedDefaultRunfiles, executable);
```

Comment:
What is the argument against creating a `RunfilesSupport.withExecutableAndArgs()` method that overrides the argv computed from the attributes of the rule? That looks simpler than plumbing another set of arguments to `RunCommand` / `TestActionBuilder` and carefully making sure that the one in `RunEnvironmentInfo` overrides the one in `RunfilesSupport`.

Reply:
@fmeum ^^